akka-actor/src/main/scala/akka/io/dns/DnsSettings.scala

```diff
@@ -44,6 +46,50 @@ private[dns] final class DnsSettings(system: ExtendedActorSystem, c: Config) {
 
   val ResolveTimeout: FiniteDuration = c.getDuration("resolve-timeout").asScala
 
+  private lazy val resolvConf: Option[ResolvConf] = {
+    val etcResolvConf = new File("/etc/resolv.conf")
+    // Avoid doing the check on Windows, no point
+    if (sys.props.get("os.name").exists(os ⇒ os.startsWith("Windows") || os.startsWith("WINDOWS"))) {
+      None
+    } else if (etcResolvConf.exists()) {
+      val parsed = ResolvConfParser.parseFile(etcResolvConf)
+      parsed match {
+        case Success(value) ⇒ Some(value)
+        case Failure(exception) ⇒
+          if (system.log.isWarningEnabled) {
+            system.log.error(exception, "Error parsing /etc/resolv.conf, ignoringto")
+          }
+          None
+      }
+    } else None
+  }
+
+  val SearchDomains: List[String] = {
+    c.getValue("search-domains").valueType() match {
+      case ConfigValueType.STRING ⇒
+        c.getString("search-domains") match {
+          case "default" ⇒ resolvConf.map(_.search).getOrElse(Nil)
+          case single    ⇒ List(single)
+        }
+      case ConfigValueType.LIST ⇒
+        c.getStringList("search-domains").asScala.toList
+      case _ ⇒ throw new IllegalArgumentException("Invalid type for search-domains. Must be a string or string list.")
+    }
+  }
```

Comment:
Is it an Akka-specific requirement to support both `STRING` and `LIST` config types? Could we _just_ make the default `[]` and require this setting to always be a `List[String]`?

Reply:
I've basically just copied the code above for handling `nameservers` config here, so it's consistent with the `nameservers` config which I think makes sense. There's actually a good reason for supporting both, we have to still read it and check the type, since that's required to read a value of `default`. Since we're reading a single string value anyway, may as well use that value if it's not `default`.